### ISSUE TITLE
Better eslint a11y rule defaults

### ISSUE THREAD

**Issue by herecydev** [gatsbyjs/gatsby#24582]
## Summary

The default [a11y eslint rules](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/eslint-config.ts) are all defined as "warn". This is okay for most rules but for the newly added "jsx-a11y/control-has-associated-label" rule, adding "warn" is making it almost unusable.

Even the following is flagged by eslint 🤔 :

```jsx
<label>lovely label<input type="text/></label>
```

The [recommended config for this rule](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/src/index.js#L64-L92) is much more sane.

Would you accept a PR to fix this?

**Comment by madalynrose** [comment]
I think we still want the strict configuration for the rule. I'm confused as to why that case is causing it to trigger though. Going by [their documentation](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md#how-do-i-resolve-this-error), your usage (aside from the missing `"`) should be fine. Do have a repo you could point me to where perhaps I could help troubleshoot? I'd prefer to fix whatever is causing the error than not warn when something breaks accessibility.

**Comment by herecydev** [comment]
> I think we still want the strict configuration for the rule. I'm confused as to why that case is causing it to trigger though. Going by [their documentation](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md#how-do-i-resolve-this-error), your usage (aside from the missing `"`) should be fine. 

Because their default (even in strict mode) is not "warn", instead it's an object that specificies some [ignored elements](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/src/index.js#L221-L247).

> I'd prefer to fix whatever is causing the error than not warn when something breaks accessibility.

100% agree but the issue as I see it is that Gatsby is taking the defaults provided by eslint-a11y and then tweaking them differently. I can't help think that Gatsby should just take what's there and map any rule "errors" to "warn", [until Gatsby v3](https://github.com/gatsbyjs/gatsby/issues/19945).

That way things don't get missed, the TODO there that is out-of-date and that would have been implemented already, etc.

Let me know if that helps?

**Comment by madalynrose** [comment]
Hi there @herecydev! Looks like they responded in the [issue you linked to](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/681) with a recommendation for making your example not trigger the warning, using for/ID-linking. Does that resolve your issue?

**Comment by herecydev** [comment]
Nope, there's nothing broken with using `<label>lovely label<input type="text"/></label>`. It passes the tests using the defaults in eslint-a11y.

If you look here in the rule code they're trying to cater for specifically the case I've mentioned: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/src/rules/control-has-associated-label.js#L58-L62

I can't stress any more how the issue is that Gatsby has **changed the default rule configuration and is causing that issue to break**

**Comment by herecydev** [comment]
@madalynrose I've made those changes.

**Comment by dmorda** [comment]
Since this change, I noticed I'm getting warnings for the rule when using in combination with the Formik <Field> tag.

**No warning**
```
<label htmlFor="phone">
   Phone
   <input type="text" id="phone" name="phone" />
</label>
```

**Get warning**
```
<label htmlFor="phone">
   Phone
   <Field type="input" id="phone" name="phone" />
</label>
```

Both approaches produce the same HTML output. Is there a way to address that?

**Comment by herecydev** [comment]
I think this case may be relevant to you: [custom component](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/control-has-associated-label.md#case-my-label-and-input-components-are-custom-components-but-i-still-want-to-require-that-they-have-an-accessible-text-label)

It looks like by supplying that to the config it will then validate the rendered input (that Formik will produce) against the label
